contracts/examples/VCPayment-v2.sol

```diff
@@ -0,0 +1,228 @@
+     */
+    mapping(bytes32 paymentDataId => PaymentData paymentData) private paymentData;
+
+    /**
+     * @dev mapping of paymentRequestId - keccak256(abi.encode(issuerId, paymentId)) => bool
+     */
+    mapping(bytes32 paymentRequestId => bool isPaid) public payments;
+
+    /**
+     * @dev mapping of issuerAddress - balance
+     */
+    mapping(address issuerAddress => uint256 balance) public issuerAddressBalance;
+
+    /**
+     * @dev owner balance
+     */
+    uint256 ownerBalance;
+
+    /**
+     * @dev list of paymentDataId - keccak256(abi.encode(issuerId, schemaHash))
+     */
+    bytes32[] internal paymentDataIds;
+
+    /**
+     * @dev Version of contract
+     */
+    string public constant VERSION = '2.0.0';
+
+    /**
+     * @dev Version of contract
+     */
+    uint256 public ownerPartPercent;
+
+    event Payment(
+        uint256 indexed issuerId,
+        string paymentId,
+        uint256 indexed schemaHash,
+        uint256 timestamp
+    );
+
+    error InvalidOwnerPartPercent(string message);
+    error InvalidWithdrawAddress(string message);
+    error PaymentError(string message);
+    error WithdrawError(string message);
+    error OwnerOrIssuerError(string message);
+
+    /**
+     * @dev Owner or issuer modifier
+     */
+    modifier ownerOrIssuer(uint256 issuerId, uint256 schemaHash) {
+        address issuerAddress = paymentData[keccak256(abi.encode(issuerId, schemaHash))]
+            .withdrawAddress;
+        if (issuerAddress != _msgSender() && owner() != _msgSender()) {
+            revert OwnerOrIssuerError('Only issuer or owner can call this function');
+        }
+        _;
+    }
+
+    constructor(uint256 _ownerPartPercent) Ownable(_msgSender()) {
+        if (ownerPartPercent > 0 && ownerPartPercent < 100) {
+            revert InvalidOwnerPartPercent('Invalid owner part percent');
+        }
+        ownerPartPercent = _ownerPartPercent;
+    }
+
+    function updateOwnerPartPercent(uint256 _ownerPartPercent) public onlyOwner {
+        if (ownerPartPercent > 0 && ownerPartPercent < 100) {
+            revert InvalidOwnerPartPercent('Invalid owner part percent');
+        }
+        ownerPartPercent = _ownerPartPercent;
+    }
+
+    function setPaymentValue(
+        uint256 issuerId,
+        uint256 schemaHash,
+        uint256 value,
+        address withdrawAddress
+    ) public onlyOwner {
+        if (withdrawAddress == address(0)) {
+            revert InvalidWithdrawAddress('Invalid withdraw address');
+        }
+        PaymentData memory newPaymentData = PaymentData(
+            issuerId,
+            schemaHash,
+            value,
+            withdrawAddress,
+            0
+        );
+        paymentDataIds.push(keccak256(abi.encode(issuerId, schemaHash)));
+        _setPaymentData(issuerId, schemaHash, newPaymentData);
+    }
+
+    function updateWithdrawAddress(
+        uint256 issuerId,
+        uint256 schemaHash,
+        address withdrawAddress
```

Comment:
If the same issuer has two different schemas, then it will switch only one paymentDataId and transfers issuer's balance. However the paymentDataId belonging to different schema will still be collect funds for old withdrawal address. Is it expected?

Reply:
Still need some answer here.